Lib/glyphsLib/builder/transformations/propagate_anchors.py

```diff
@@ -299,6 +300,7 @@ def origin_adjusted_anchors(anchors: list[GSAnchor]) -> Iterable[GSAnchor]:
         GSAnchor(
             name=a.name,
             position=Point(a.position.x - origin.x, a.position.y - origin.y),
+            userData=deepcopy(a.userData),
```

Comment:
in any case, I think just a shallow `dicgt(a.userData)` should be sufficient for this, no need to deep copy or customise deepcopying of the UserDataProxy class